server/routes/scan/ScanBarcodeController.ts

```diff
@@ -99,14 +101,7 @@ export default class ScanBarcodeController {
         req.session.barcodeEntryForm.createdBy = checkBarcodeResponse.createdBy
       })
       .catch(errorResponse => {
-        const errorType = errorResponse.data?.errorCode?.code
```

Comment:
This catch block was getting messy (and was untested!) and was only going to get messier (and more untested!) with the logic to get the prison name for Duplicate barcode scans.
So I refactored it all out into a new class which I've lovingly called a 'Error Response Mapper', including writing unit tests for it 👍